File: 21주차/성격유형검사하기.py
def solution(survey,choices):
    answer = ''
    a = {'R':0, 'T':0}
    b = {'C': 0, 'F':0}
    c = {'J':0, 'M':0}
    d = {'A':0, 'N':0}
    
    for i,v in enumerate(survey):
        l = v[0]
        r = v[1]
        if l in d:
            if choices[i] > 4:
                d[r] += abs(choices[i] - 4)
            else:
                d[l] += abs(choices[i] - 4)
        elif l in a:
            if choices[i] > 4:
                a[r] += abs(choices[i] - 4)
            else:
                a[l] += abs(choices[i] - 4)
        elif l in b:
            if choices[i] > 4:
                b[r] += abs(choices[i] - 4)
            else:
                b[l] += abs(choices[i] - 4)
        elif l in c:
            if choices[i] > 4:
                c[r] += abs(choices[i] - 4)
            else:
                c[l] += abs(choices[i] - 4)
    a= list(a.items())
    a.sort(key=lambda x:(-x[1],x[0]))
    
    b= list(b.items())
    b.sort(key=lambda x:(-x[1],x[0]))
    
    c= list(c.items())
    c.sort(key=lambda x:(-x[1],x[0]))
    
    d= list(d.items())
    d.sort(key=lambda x:(-x[1],x[0]))
    
    return a[0][0] + b[0][0] + c[0][0] + d[0][0]

File: 49주차/성격유형검사하기.js
function solution(survey, choices) {
  let dict = {};
  const types = ["RT", "CF", "JM", "AN"];

  types.forEach((v) => v.split("").forEach((v) => (dict[v] = 0))); //딕셔너리에 초기화.

  choices.forEach((v, i) => {
    dict[v > 4 ? survey[i][1] : survey[i][0]] += Math.abs(v - 4);
  }); // 567 은 오른쪽 유형에 점수 획득, 123은 왼쪽 유형에 점수 획득.

  // 점수가 더 높은 유형이 선택되고, 같다면 사전순. 이미 types에서 사전순이므로, a반환.
  return types.map(([a, b]) => (dict[b] > dict[a] ? b : a)).join("");
}
// 프로그래머스 카카오 인턴십문제.
